VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -66,16 +72,53 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
                 _logger.LogException($"Failed to start process based host with instance number {instanceNumber}", exception);
                 sessionHostManager.RemoveSessionHost(sessionHostUniqueId);
                 sessionHost = null;
+                string exceptionMessage = exception.ToString();
+                CollectLogs(logFolderPathOnVm, exceptionMessage);
             }
 
             return Task.FromResult(sessionHost);
         }
 
-        public override Task CollectLogs(string processId, string logsFolder, ISessionHostManager sessionHostManager)
+        public override Task CollectLogs(string id, string logsFolder, ISessionHostManager sessionHostManager)
+        {
+            return Task.CompletedTask;
+        }
+
+        public override Task CollectLogs(string logsFolder, string exceptionMessage)
         {
             // The game server is free to read the env variable for log folder and write all output to a file in that folder.
             // If required, we can add action handlers (see SystemOperations.RunProcess for example).
             // However, keeping the file handle around can be tricky.
+            if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
+            {
+                try
+                {
+                    _logger.LogVerbose("Collecting logs for failed start game process.");
+                    string destinationFileName = Path.Combine(logsFolder, ConsoleLogCaptureFileName);
+
+                    using (StreamWriter sw = File.AppendText(destinationFileName))
+                    {
+                        try
+                        {
+                            sw.WriteLine("{0} {1}", DateTime.Now.ToLongTimeString(), DateTime.Now.ToLongDateString());
+                            sw.WriteLine($"{exceptionMessage}");
+                        }
+                        catch (Exception ex)
+                        {
+                            sw.WriteLine(ex.ToString());
```

Comment:
Is this try-catch necessary? It looks like the only thing that could fail in it is "sw.WriteLine", which is exactly what we do in the Catch block so it's just going to fail again